awx/main/analytics/collectors.py

```diff
@@ -344,39 +359,50 @@ def _copy_table(table, query, path):
-                         main_jobevent.uuid,
-                         main_jobevent.parent_uuid,
-                         main_jobevent.event, 
+        # TODO: conditional job_created based on if the column exists or not in the table
+        # {tbl}.job_created,
+        return f'''COPY (SELECT {tbl}.id,
+                         {tbl}.created,
+                         {tbl}.modified,
+                         {tbl}.uuid,
+                         {tbl}.parent_uuid,
+                         {tbl}.event,
                          {event_data}->'task_action' AS task_action,
                          (CASE WHEN event = 'playbook_on_stats' THEN event_data END) as playbook_on_stats,
-                         main_jobevent.failed, 
-                         main_jobevent.changed, 
-                         main_jobevent.playbook, 
-                         main_jobevent.play,
-                         main_jobevent.task,
-                         main_jobevent.role, 
-                         main_jobevent.job_id, 
-                         main_jobevent.host_id, 
-                         main_jobevent.host_name,
+                         {tbl}.failed,
+                         {tbl}.changed,
+                         {tbl}.playbook,
+                         {tbl}.play,
+                         {tbl}.task,
+                         {tbl}.role,
+                         {tbl}.job_id,
+                         {tbl}.host_id,
+                         {tbl}.host_name,
                          CAST({event_data}->>'start' AS TIMESTAMP WITH TIME ZONE) AS start,
                          CAST({event_data}->>'end' AS TIMESTAMP WITH TIME ZONE) AS end,
                          {event_data}->'duration' AS duration,
                          {event_data}->'res'->'warnings' AS warnings,
                          {event_data}->'res'->'deprecations' AS deprecations
-                         FROM main_jobevent 
-                         WHERE (main_jobevent.id > {since} AND main_jobevent.id <= {until})
-                         ORDER BY main_jobevent.id ASC) TO STDOUT WITH CSV HEADER'''
+                         FROM {tbl}
+                         WHERE ({tbl}.id > {since} AND {tbl}.id <= {until})
+                         ORDER BY {tbl}.id ASC) TO STDOUT WITH CSV HEADER'''
 
     try:
-        return _copy_table(table='events', query=query("main_jobevent.event_data::json"), path=full_path)
+        return _copy_table(table='events', query=query(f"{tbl}.event_data::json"), path=full_path)
     except UntranslatableCharacter:
-        return _copy_table(table='events', query=query("replace(main_jobevent.event_data::text, '\\u0000', '')::json"), path=full_path)
+        return _copy_table(table='events', query=query(f"replace({tbl}.event_data::text, '\\u0000', '')::json"), path=full_path)
+
+
+@register('events_table', '1.2', format='csv', description=_('Automation task records'), expensive=events_slicing_unpartitioned)
```

Comment:
This is an intriguing approach, splitting off the legacy collection like this.  I think I might like it.

Reply:
Wait, no, there's a problem.  Each one will clobber the other's bookkeeping, since they both fall under `'events_table'`.

... or will it?  I have logic in place that ensures we don't ever rewind the bookkeeping to an earlier point, maybe that's enough?

Reply:
Ok, I think I'm back to 'I like it'.  This should do the right thing, due to the "don't rewind" thing mentioned above, and this will make it super-easy to chop out support for the legacy un-partition if and when we drop it.